### ISSUE TITLE
ViewPager's selectedIndex prop does not change children <View/>

### ISSUE THREAD

**Issue by cereme** [akveo/react-native-ui-kitten#547]
## Issue type

**I'm submitting a ...**  (check one with "x")

* [x] bug report
* [ ] feature request

### Issue description

**Current behavior:**
ViewPager's View does not change when selectedIndex props change.
Swiping changes selectedIndex properly.
(Please check the gif below)

**Expected behavior:**
When change selectedIndex props(with Hooks/setState), ViewPager's View changes

**Steps to reproduce:**
1. set ViewPager's selectedIndex prop as ```state```
2. invoke state change(with Hooks/setState) the selectedIndex prop's state
3. ViewPager's View does not chage

**Related code:**
```javascript
import React, { useState } from 'react';
import { View } from 'react-native';
import { Text, ViewPager, Button } from 'react-native-ui-kitten';

export default function MainPage(){
  const [pageIndex, setIndex] = useState(0);
  const tabNames = ['a', 'b', 'c', 'd', 'e'];
  return(
    <View style={{flex: 1}}>
      <View style={{flex: 0}}>
        <Button onPress={()=>{setIndex(pageIndex - 1)}}> - </Button>
        <Button onPress={()=>{setIndex(pageIndex + 1)}}> + </Button>
      </View>
      <ViewPager style={{flex: 1}}
        selectedIndex={pageIndex}
        onSelect={(number)=>{setIndex(number);}}>
          {tabNames.map((item, index) => (
            <View key={index}>
              <Text>{item}, {index}</Text>
            </View>
          ))}
      </ViewPager>
      <View style={{flex: 0}}>
        <Text>{pageIndex}</Text>
      </View>
    </View>
  )
};
```
![viewpager](https://user-images.githubusercontent.com/19284878/61961929-678a0600-b003-11e9-8da9-c2a7ec56d3d3.gif)
Swiping works well, though.
I tried setState instead of hooks, the problem was same.

### Other information:

**OS, device, package version**
```
Android 9.0
React native 0.60.4
react-native-ui-kitten 4.1.0
```

**Comment by artyorsh** [comment]
Thanks for reporting this. Will take a look👍

**Comment by artyorsh** [comment]
@cereme we've got a working example in master branch. Will be published in 4.2

**Comment by Alta1re** [comment]
Still have exactly the same issue on V5.1.1, on both platforms (iOS, Android).
Swiping works fine. External change of selectedIndex does not change the view.
